src/styles/main.scss

```diff
@@ -0,0 +1,95 @@
+$day-size: 100px;
+$transition-time: 0.5s;
+$calendar-width: 7 * ($day-size + 1px);
+$days-of-week: (mon, tue, wed, thu, fri, sat, sun);
+$calendar-padding: 20px;
+$bg-color: #eee;
+
+body {
+  margin: 0;
+  height: 100vh;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.calendar {
+  counter-reset: day-counter;
+  display: flex;
+  flex-wrap: wrap;
+  width: $calendar-width;
+  padding: $calendar-padding;
+  gap: 1px;
+
+  &__day {
+    box-sizing: border-box;
+    width: $day-size;
+    height: $day-size;
+    background-color: $bg-color;
+    border: 1px solid black;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    font-family: Arial, sans-serif;
+    font-size: 30px;
+    line-height: 30px;
+    text-align: center;
+    position: relative;
+    transition:
+      transform 0.5s,
+      background-color 0.5s;
+
+    &::before {
+      content: '';
+      position: absolute;
+      bottom: ($day-size - 2 * 1px) / 2;
+      right: ($day-size - 3 * 1px) / 2;
+      transform: translate(50%, 50%);
+      font-family: Arial, sans-serif;
+      font-size: 30px;
+      line-height: 30px;
+      text-align: center;
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      width: $day-size;
+      height: $day-size;
+      counter-increment: day-counter;
```

Comment:
Styles order is important
https://9elements.com/css-rule-order/
Check everywhere